File: robot_examples/haskell-servant/rest_api/lib/Model/EventMissile.hs
{-# LANGUAGE DataKinds #-}
{-# LANGUAGE DeriveGeneric #-}
{-# LANGUAGE TypeOperators #-}
{-# LANGUAGE OverloadedStrings #-}
{-# LANGUAGE GeneralizedNewtypeDeriving #-}

module Model.EventMissile
    ( EventMissile (..)
    ) where

import Data.Aeson
import GHC.Generics
import Test.QuickCheck
import Model.Event
import Model.Number
import Model.RobotInfo


data EventMissile = EventMissile
    { eventType :: Number
    , activationTime :: Float
    , robot :: RobotInfo
    , direction :: Float
    , distance :: Float
    , speed :: Float
    } deriving (Show, Eq, Generic)

instance FromJSON EventMissile
instance ToJSON EventMissile
instance Arbitrary EventMissile where
    arbitrary = EventMissile <$> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary <*> arbitrary

File: robot_examples/go/rest_api/swagger/EventMissile.go
package swagger

import (
)

type EventMissile struct {
    EventType  Number  `json:"eventType,omitempty"`
    ActivationTime  float32  `json:"activationTime,omitempty"`
    Robot  RobotInfo  `json:"robot,omitempty"`
    Direction  float32  `json:"direction,omitempty"`
    Distance  float32  `json:"distance,omitempty"`
    Speed  float32  `json:"speed,omitempty"`
    
}
